src/clients/prometheus.py

```diff
@@ -27,6 +29,33 @@ def __init__(self, config: PrometheusConfig):
             max_parallel_queries=config.max_parallel_queries,
         )
 
+    def create_auth(self) -> Optional[PrometheusAuth]:
+        """Create the appropriate authentication object based on the configuration."""
+        auth_config = self.config.auth
+        if not auth_config:
+            return None
+        try:
+            if auth_config.username and auth_config.password:
+                return PrometheusAuth(
+                    username=auth_config.username, password=auth_config.password
+                )
+            elif auth_config.token:
+                return PrometheusAuth(token=auth_config.token)
+            elif auth_config.cert_path and auth_config.key_path:
+                return PrometheusAuth(ca_cert=auth_config.cert_path)
+            elif (
+                auth_config.gcloud_service_account_path
+                and auth_config.gcloud_target_principal
+            ):
+                return GCPPrometheusAuth(
+                    service_account_file=auth_config.gcloud_service_account_path,
+                    target_principal=auth_config.gcloud_target_principal,
+                )
+            return None
+        except Exception as e:
+            logger.error(f"Failed to create auth: {str(e)}")
+            return None
```

Comment:
it's really defensive, and i is ever possible?

if you really, really, really need and you really really reall reall think it's a good idea, having decorator like:

```
def none_on_exception(message_format="Failed on exception %s", base_exception: type[Exception] = Exception):
    ...
```
will be much nicer - there will be no massive `try ... catch` in the code